integration/pull_test.go

```diff
@@ -148,22 +147,15 @@ func TestOptimizeConsistentSociArtifact(t *testing.T) {
 
 func TestLazyPullWithSparseIndex(t *testing.T) {
 	regConfig := newRegistryConfig()
```

Comment:
In some places you moved this into `newShellWithRegistry` instead of putting it into a variable. Was it a conscious effort not to do it in some places like this?

Reply:
I left unchanged the locations where regConfig is used repeatedly instead of just once. See lines below including `copyImage(sh, dockerhub(imageName), regConfig.mirror(imageName))` and `sh.X("ctr", "i", "pull", "--user", regConfig.creds(), image)`

Reply:
It does seem a little silly that we are so often instanciating a registryConfig just to access its default properties, but changing that seems like a bigger refactor.

Reply:
Ah, right. `newRegistryConfig` isn't deterministic, it generates an XID that's used in the hostname when configuring the registry.